tomviz/HistogramWidget.cxx

```diff
@@ -516,7 +517,13 @@ void HistogramWidget::onSaveToPresetClicked()
 
 void HistogramWidget::onPresetClicked()
 {
-  showPresetDialog(this, nullptr);
+  if (m_presetDialog == nullptr) {
+    m_presetDialog = new PresetDialog(this);
+  }
+  QObject::connect(m_presetDialog, SIGNAL(applyPreset(const Json::Value&)),
+      this, SLOT(applyCurrentPreset()));
```

Comment:
This would probably be better placed within the `if()` block above. Otherwise, an extra connection will be made every time `onPresetClicked()` is called, and `applyCurrentPreset()` will be called the number of times it was connected (so if `onPresetClicked()` gets called 3 times, for instance, the third time it is called, `applyCurrentPreset()` will be called 3 times for a single `applyPreset` signal).

Also, this is using the old signal and slot syntax in Qt. You don't necessarily need to change it to the new syntax, but I think it can be useful for a few reasons, including that the new syntax will give a compile error when improperly connected, whereas the old syntax gives a runtime error. See [here](https://wiki.qt.io/New_Signal_Slot_Syntax) for more information. I think this would be changed to this if you went with the new syntax:
```
QObject::connect(m_presetDialog, &PresetDialog::applyPreset, this, &HistogramWidget::applyCurrentPreset);`
```

Reply:
I think we should be happy with it either way in Tomviz, and not devote too much time/effort to enforcing one or the other style. Happy to prefer the new signatures, but I wouldn't worry about changing it either. It is a little nicer for failing more predictably using the compiler to create connections.